packages/integration-sdk-cli/src/commands/diff.ts

```diff
@@ -0,0 +1,197 @@
+import { createCommand } from 'commander';
+import fs from 'fs';
+import { diffString } from 'json-diff';
+
+declare module 'json-diff' {
+  /**
+   * The exported types from @types/json-diff exclude an optional fourth parameter, `diffOptions`,
+   * that allows the user to pass the `keysOnly` flag. Sadly, based on testing, it seems that
+   * `keysOnly` does not work under certain circumstances, particularly with diffed arrays.
+   *
+   * Method signature as exported from @types/json-diff:
+   *   function diffString(obj1: unknown, obj2: unknown, colorizeOptions?: ColorizeOptions): string;
+   */
+  function diffString(
+    obj1: unknown,
+    obj2: unknown,
+    colorizeOptions?: { color?: boolean },
+    diffOptions?: { keysOnly?: boolean },
+  ): string;
+}
+
+export function diff() {
+  return createCommand('diff')
+    .storeOptionsAsProperties()
+    .arguments('<oldExportPath> <newExportPath>')
+    .description(
+      'Compare two datasets downloaded from JupiterOne using: \n\n' +
+        '\t find * with _integrationInstanceId="abc-123" that RELATES TO * return tree\n\n',
```

Comment:
Keen observers will notice that this J1QL should _actually_ use an optional relationship, in order to capture any entities that have no edges:
```
FIND * with _integrationInstanceId='integration-instance-id' (THAT RELATES TO *)? RETURN TREE
```

However, I tested this in dev, and I'd say that the non-optional traversal query may take ~5 seconds, and the optional traversal query may take well over a minute. For that reason, I've chosen _not_ to use the optional traversal query in this description.

Anyone disagree? Think that this should include the optional traversal?

Reply:
Very good find @ndowmon. I had originally assumed that all entities are related to something, essentially making the `Account` entity of the integration the root node. However, that's certainly not true all the time. Perhaps we could use two queries to make it faster and to separate entities from relationships?

1. `FIND * WITH _integrationInstanceId='integration-instance-id'`
2. `FIND * WITH _integrationInstanceId='integration-instance-id' THAT RELATES TO AS r * RETURN r.*`

Otherwise, perhaps it's just fine to wait a minute since we're looking to validate data, not run the query a lot of times and wait.